flexmeasures/data/models/planning/storage.py

```diff
@@ -172,7 +180,191 @@ def schedule(
             battery_schedule = ems_schedule[0]
 
         # Round schedule
-        if round_to_decimals:
-            battery_schedule = battery_schedule.round(round_to_decimals)
+        if self.round_to_decimals:
+            battery_schedule = battery_schedule.round(self.round_to_decimals)
 
         return battery_schedule
+
+    def persist_flex_model(self):
+        """Store new soc info as GenericAsset attributes"""
+        self.sensor.generic_asset.set_attribute("soc_datetime", self.start.isoformat())
+        if self.flex_model.get("soc_unit") == "kWh":
+            self.sensor.generic_asset.set_attribute(
+                "soc_in_mwh", self.flex_model["soc_at_start"] / 1000
+            )
+        else:
+            self.sensor.generic_asset.set_attribute(
+                "soc_in_mwh", self.flex_model["soc_at_start"]
+            )
+
+    def inspect_flex_config(self):
```

Comment:
Maybe just a note for later: we use a Marshmallow schema in this function, which by now does not only do validation, but also deserialization (which is what changes the flex config fields with dash notation to the flex config variables with underscore notation). This is already disclosed in the docstring, but I believe the function name could bring that to light better, because it's more than an inspection at this point. Maybe `deserialize_flex_config`. The function also fills in defaults (if needed), but this can be done in the schema, too, using [`load_default`](https://marshmallow.readthedocs.io/en/stable/quickstart.html#specifying-defaults).